server-spi-private/src/main/java/org/keycloak/models/utils/TimeBasedOTP.java

```diff
@@ -93,6 +98,13 @@ public boolean validateTOTP(String token, byte[] secret) {
         return false;
     }
 
+    /**
+     * maps 0, 1, 2, 3, 4, 5, 6, 7, ... to 0, -1, 1, -2, 2, -3, 3, ...
+     */
+    public static long clockSkewIndexToDelta(int idx) {
```

Comment:
I'm not sure about the access modifier for this. When this method/function is revealed and can be accessed everywhere, the behavior should be described/defined better; is allowed to provide a negative index? If the function is private, it's not so important to do that with a such effort, because we are sure the index will never be a negative number. @thomasdarimont WDYT ?

IMO there should be, at least, present some test cases for testing this function in order to ensure the expected behavior.

Reply:
@mabartos I'll just make it private :)